### ISSUE TITLE
Extend documentation and samples

### ISSUE THREAD

**Issue by martincostello** [martincostello/xunit-logging#157]
Based on #156, the documentation could be expanded (and the samples extended) to show usage of the logger with a test fixture, rather than just in a standalone test.

Also consider an example that supports parallel execution of tests without bleeding the logger between them.

**Comment by mderriey** [comment]
> to show usage of the logger with a test fixture

I'd be super interested in this.

Usually, I set up a collection fixture where the service collection / test server is created once, then all test classes use that same instance to run tests.

My understanding is that `ITestOutputHelper` is only injectable in test classes constructors, and I don't think we can "inject" loggers after the service provider is built.

Am I missing something?

**Comment by martincostello** [comment]
That's correct, but it's possible to still use it with a shared fixture with the use of the `ITestOutputHelperAccessor` interface and a bit of boilerplate in a base class. This just isn't explicitly spelled out in the docs here yet 😃.

From an example in one of my own projects.

1. The shared fixture implements `ITestOutputHelper` and [provides a mechanism](https://github.com/martincostello/website/blob/3a2b063d4cd1bc99f8e355ffdb402e44af446ff4/tests/Website.Tests/Integration/TestServerFixture.cs#L29-L55) to set and clear the `ITestOutputHelper`.
2. A base class for the tests [sets and clears the output helper](https://github.com/martincostello/website/blob/3a2b063d4cd1bc99f8e355ffdb402e44af446ff4/tests/Website.Tests/Integration/IntegrationTest.cs#L19-L63).
3. Individual test classes [pass through the output helper](https://github.com/martincostello/website/blob/3a2b063d4cd1bc99f8e355ffdb402e44af446ff4/tests/Website.Tests/Integration/ResourceTests.cs#L19-L22) to the base class.

xunit then does the rest of the heavy lifting for you. The only downside (which is inherent to how xunit works) is that you can't have multiple tests running in parallel with the same fixture, as they'll otherwise be fighting over the current output helper. Maybe you could wrap it and have multiple helpers and cycle them based on the current thread/fixture or something and delegate it via a reference to overcome that, but it's not something I've tried.

**Comment by martincostello** [comment]
All levels except `None` are logged by default unless you provide a custom [Filter method](https://github.com/martincostello/xunit-logging/blob/f01740ba004a37ffe1a9c94341c6383ff57689a3/src/Logging.XUnit/XUnitLogger.cs#L64-L74):

https://github.com/martincostello/xunit-logging/blob/f01740ba004a37ffe1a9c94341c6383ff57689a3/src/Logging.XUnit/XUnitLogger.cs#L103-L111

My guess would be your application is configured to not log whatever you're looking for at Debug or Trace.
